spec/controllers/orders_controller_spec.rb

```diff
@@ -2,46 +2,59 @@
 
 RSpec.describe OrdersController, type: :controller do
   describe "GET #new" do
-    subject { get :new, event_id: event.id }
-
-    context "this event has expired" do
-      let!(:event) { create(:expired_event) }
+    context "this event does not exist" do
+      before do
+        get :new, event_id: INVALID_ID
+      end
 
-      it "is successful" do
-        subject
-        expect(response).to be_success
-        expect(assigns(:event)).to eq event
+      it "redirects to Home page and shows error message" do
+        expect(response).to redirect_to root_path
+        expect(flash[:alert]).to eq EVENT_NOT_FOUND
       end
     end
 
-    context "this event has not expired" do
-      let(:event) { create(:event) }
-
-      context "user has already signed in" do
-        login
-
-        let(:order) { double }
-
-        before do
-          expect(Order).to receive(:build_from_event).and_return order
-        end
+    context "this event exists" do
+      subject { get :new, event_id: event.id }
+      
+      context "this event has expired" do
+        let!(:event) { create(:expired_event) }
 
-        it "is successful and creates new order with some order items based on the ticket types of this event" do
+        it "redirects to Home page and shows error message" do
           subject
-          expect(response).to be_success
-          expect(assigns(:event)).to eq event
-          expect(assigns(:order)).to eq order
+          expect(response).to redirect_to root_path
+          expect(flash[:alert]).to eq EXPIRED_EVENT
         end
       end
 
-      context "user has not signed in yet" do
-        before do
-          expect(controller).to receive(:store_location_and_require_login).and_call_original
-          subject
+      context "this event has not expired" do
+        let(:event) { create(:event) }
+
+        context "user has already signed in" do
+          login
+
+          let(:order) { create(:order) }
+
+          before do
+            expect(Order).to receive(:build_from_event).and_return order
+          end
+
+          it "is successful and creates new order with some order items based on the ticket types of this event" do
+            subject
+            expect(response).to be_success
+            expect(assigns(:event)).to eq event
+            expect(assigns(:order)).to eq order
+          end
         end
 
-        it "redirects to Sign in page" do
-          expect(response).to redirect_to login_path
+        context "user has not signed in yet" do
```

Comment:
Okay this may take time but we should find a way to refactor controller test for unauthorized actions.